src/filter/GaussianFilter.js

```diff
@@ -0,0 +1,58 @@
+import IJ from '../ij';
+
+export default function gaussian_filter(n, sigma){
+	var cols = this.width;
+	var new_img = this.clone();
+	var kernel = get_gaussian_kernel(n, sigma);
+	var temp = cols*n; //variable for checking the image border
+	for(var p = cols*n+n; p < this.data.length - cols*n; p++){
+		if(p == temp+cols-n){
+			p += 2*n - 1;
+		}else{
+			new_img.data[p] = convolution(p, this.data, kernel);
+		}
+	}
+}
+
+function get_gaussian_kernel(n, sigma){
+	var kernel = [];
+	var sigma2 = 2*(sigma*sigma);//2*sigma^2
+	var sum = 0;
+	for(var y = -n; y <= n; y++){
+		for(var x = -n; x <= n; x++){
+			var value = -((x*x) + (y*y))/sigma2;
+			value = Math.pow(Math.E, value);
+			sum += value;
+			kernel.push(value);
+		}
+	}
+
+	for(var i = 0; i < kernel.length; i++){
+		kernel[i] /= sum;
+	}
+	return kernel;
+}
+
+function convolution(p, data, kernel){
+	var result = 0;
+	var n = (int)(Math.sqrt(kernel.length) - 1)/2;
```

Comment:
(int) does not exists in javascript. In javascript there is only "Number" that are float 64. In the number 53 bits is devoted for the integer part of the Number (however only 32 bits can bit use for bitwise operations).
If you want just the integer part of your Number you have many possibilities:
* Math.floor
* Math.ceil
* Math.round
* 123.123 | 0 : this last trick applies a bitwise operation of a decimal number and the first step is to keep 32 bits of the integer part of the number http://www.ecma-international.org/ecma-262/6.0/#sec-toint32